src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -104,6 +104,56 @@ export const handleGetAdminForm: RequestHandler<{ formId: string }> = (
   )
 }
 
+/**
+ * Handler for GET /:formId/adminform/preview.
+ * @security session
+ *
+ * @returns 200 with form with private details scrubbed for previewing if user has read permissions
+ * @returns 403 when user does not have permissions to access form
+ * @returns 404 when form cannot be found
+ * @returns 410 when form is archived
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const handlePreviewAdminForm: RequestHandler<{ formId: string }> = (
+  req,
+  res,
+) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+  return (
+    // Step 1: Retrieve currently logged in user.
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Check whether user has read permissions to form
+        AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Read,
+        }),
+      )
+      // Step 3: Remove private details from form for previewing.
+      .map(removePrivateDetailsFromForm)
```

Comment:
technically not part of this PR, but I just wanted to point out that this is probably better modeled as an instance method on each form object, so that one can call `form.getPublicView()`.

Reply:
it'll also get rid of the awkward util file that always contains functions that we're not sure where else to put

Reply:
I'll update it in this PR

Reply:
ok, will approve first